airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.cdk.core.operation.Operation
+import io.airbyte.cdk.core.operation.OperationType
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.util.ApmTraceUtils
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Value
+import jakarta.inject.Inject
+import jakarta.inject.Named
+import java.nio.file.Path
+import java.util.Optional
+import java.util.function.Consumer
+import picocli.CommandLine
+
+private val logger = KotlinLogging.logger {}
+
+/** CLI implementation that invokes the requested operation for the connector. */
+@CommandLine.Command(
+    name = "airbyte-connector",
+    description = ["Executes an Airbyte connector"],
+    mixinStandardHelpOptions = true,
+    header =
+        [
+            "@|magenta     ___    _      __          __       |@",
+            "@|magenta    /   |  (_)____/ /_  __  __/ /____   |@",
+            "@|magenta   / /| | / / ___/ __ \\/ / / / __/ _   |@",
+            "@|magenta  / ___ |/ / /  / /_/ / /_/ / /_/  __/  |@",
+            "@|magenta /_/  |_/_/_/  /_.___/\\__, /\\__/\\___/|@",
+            "@|magenta                    /____/              |@",
+        ],
+)
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION", "NP_OPTIONAL_RETURN_NULL"],
+    justification = "Uses dependency injection",
```

Comment:
does that mean we'll need those suppresswarnings everywhere we use DI?

Reply:
@stephane-airbyte No, unless we use `@Inject` everywhere, which is not what I recommend.  Instead, we will mark classes as `@Singleton` and then just include the dependencies in the constructor without any annotations.  In that case, spotbugs does not identify an issue.

Reply:
I should note that we have to use `@Inject` here because of PicoCLI.